packages/ui-signer/src/Extrinsic.tsx

```diff
@@ -3,32 +3,52 @@
 // of the ISC license. See the LICENSE file for details.
 
 import { I18nProps } from '@polkadot/ui-app/types';
-import { QueueTx } from './types';
+import { RxFees } from '@polkadot/api-observable/types';
+import { Fees, QueueTx } from './types';
 
 import BN from 'bn.js';
 import React from 'react';
 import { Trans } from 'react-i18next';
-import { Method } from '@polkadot/types';
+import { Balance, Method } from '@polkadot/types';
 import { Call, Modal } from '@polkadot/ui-app/index';
 import IdentityIcon from '@polkadot/ui-react/IdentityIcon';
+import withMulti from '@polkadot/ui-react-rx/with/multi';
+import withObservable from '@polkadot/ui-react-rx/with/observable';
 
+import FeeDisplay from './Fees';
 import translate from './translate';
 
 type Props = I18nProps & {
   children?: React.ReactNode,
-  value: QueueTx
+  onChangeFees: (hasAvailable: boolean) => void,
+  value: QueueTx,
+  fees: RxFees
 };
 
 class Transaction extends React.PureComponent<Props> {
   render () {
-    const { children, t, value: { accountId, accountNonce = new BN(0), extrinsic } } = this.props;
+    const { children, fees, style, t, value: { accountId, accountNonce, extrinsic } } = this.props;
+
+    const nonce = accountNonce
+      ? accountNonce
+      : new BN(0);
 
     if (!extrinsic) {
       return null;
     }
 
+    const methodInstance = new Method(extrinsic, extrinsic.meta);
+    const amount = methodInstance.args[1].raw;
+    const recipientId = methodInstance.args[0].raw.toString();
+
     const { method, section } = Method.findFunction(extrinsic.callIndex);
 
+    const isTransfer = fees && fees.transferFee && section === 'balances' && method === 'transfer';
+
+    if (isTransfer) {
+      fees.transferFee = new Balance(0);
```

Comment:
I'm not sure I get this - `transferFee` is from the state, we cannot assign it? Especially not for transfer where it needs to use the on-chain value?

Reply:
Agreed, I'm surprised it didn't result in a linting error.

Reply:
Do you actually get null/undefined/something-invalid transferFee through? If so, there is an issue in api-obvervable.

Reply:
Yes, `transferFee` is `null`, so it's not being set to the default value of `new Balance(0)` further up when destructuring from props since that'd only work if `transferFee` was `undefined`, hence the reason why I'm trying to set it to `new Balance(0)` here instead. I'll have a look into it further.